src/types.ts

```diff
@@ -3,7 +3,7 @@ import type { HttpError } from "http-errors";
 
 export type SameSiteType = boolean | "lax" | "strict" | "none";
 export type TokenRetriever = (req: Request) => string | null | undefined;
-export type DoubleCsrfCookieOptions = Omit<CookieOptions, "httpOnly">;
+export type CsrfTokenCookieOverrides = Omit<CookieOptions, "signed">;
```

Comment:
just wondering, why not letting override the signed param?

Reply:
This is only for the `generateToken` call, because signed is already configured on the middleware level, because of this part of the code here:

```
  const getCsrfCookieFromRequest = remainingCookieOptions.signed
    ? (req: Request) => req.signedCookies[cookieName] as string
    : (req: Request) => req.cookies[cookieName] as string;
```

But you're still right, this logic can be moved directly into the generateToken call itself.

Reply:
You are right, didn't think of that. Makes sense to only have either one or the other and enforce that on initialization of the library. I don't think anyone would ever use signed and non signed ones😂